projects/step-frontend/src/lib/modules/timeseries/modules/_common/types/time-series/time-series-context.ts

```diff
@@ -69,6 +75,34 @@ export class TimeSeriesContext {
     this.chartsLockedState$.complete();
   }
 
+  /**
+   * This method will not trigger change event, only if there are real changes in at least one attribute was added/removed
+   * @param attributes
```

Comment:
Putting a comment to a complex function can make sense, defining empty params is not helpful though and should be removed